staging/src/k8s.io/component-base/metrics/testutil/metrics.go

```diff
@@ -192,23 +195,35 @@ func GetHistogramFromGatherer(gatherer metrics.Gatherer, metricName string) (His
 	}
 
 	if metricFamily == nil {
-		return Histogram{}, fmt.Errorf("metric %q not found", metricName)
-	}
-
-	if metricFamily.GetMetric() == nil {
-		return Histogram{}, fmt.Errorf("metric %q is empty", metricName)
+		return metricsMap, fmt.Errorf("metric %q not found", metricName)
 	}
 
 	if len(metricFamily.GetMetric()) == 0 {
-		return Histogram{}, fmt.Errorf("metric %q is empty", metricName)
-	}
-
-	return Histogram{
+		return metricsMap, fmt.Errorf("metric %q is empty", metricName)
+	}
+
+	metrics := metricFamily.GetMetric()
+	metricsMap = make(map[string]Histogram)
+	if metricName == frameworkExtensionPointDurationMetricName {
+		for _, metric := range metrics {
+			for _, label := range metric.GetLabel() {
+				if label.GetName() == ExtensionPointLabelName {
+					name := fmt.Sprintf("%s/%s", frameworkExtensionPointDurationMetricName, label.GetValue())
+					metricsMap[name] = Histogram{
+						metric.GetHistogram(),
+					}
+				}
+			}
+		}
+	} else {
 		// Histograms are stored under the first index (based on observation).
 		// Given there's only one histogram registered per each metric name, accessing
 		// the first index is sufficient.
-		metricFamily.GetMetric()[0].GetHistogram(),
-	}, nil
+		metricsMap[metricName] = Histogram{
+			metrics[0].GetHistogram(),
```

Comment:
The line doesn't seems correct as well when we want to enable the post filter metrics data for example,  I will come back to check in the following PR.

Reply:
Fixed here: https://github.com/kubernetes/kubernetes/pull/96447